doc/articles/features/windows-storage-pickers.md

```diff
@@ -0,0 +1,228 @@
+
+### Download/upload pickers
+
+In case the **FIle System Access API** is not available in the browser, Uno Platform also offers a fallback to "download" and "upload" experiences.
+
+For "upload" picker, the browser triggers a file picker dialog and Uno Platform then copies the selected files into temporary storage of the app. The `StorageFile` instance you recieve is private for your application and the changes are then not reflected in the original file. To save the changes, you need to trigger the "download picker".
+
+For download picker, the experience requires the use of `CachedFileManager`:
+
+```c#
+var savePicker = new FileSavePicker():
+savePicker.SuggestedStartLocation = PickerLocationId.DocumentsLibrary;
+savePicker.FileTypeChoices.Add("Text file", new List<string>() { ".txt" });
+savePicker.SuggestedFileName = "New Document";
+// for download picker, no dialog is actually triggered here
+// and a temporary file is returned immediately.
+var file = await savePicker.PickSaveFileAsync();
+CachedFileManager.DeferUpdates(file);
+// write the file
+await FileIO.WriteTextAsync(file, "Hello, world!");
+// this starts the download process of the browser
+await CachedFileManager.CompleteUpdatesAsync(file);
+```
+
+### Choosing supported type of pickers
+
+By default, Uno Platform attempts to use File System Access API and falls back to download/upload pickers if not available. To control this behavior, you can use `WinRTFeatureConfiguration`:
+
+```c#
+#if __WASM__
+Uno.WinRTFeatureConfiguration.Storage.Pickers.WasmConfiguration = 
+    WasmPickerConfiguration.FileSystemAccessApiWithFallback;
+#endif
+```
+
+The allowed values for the configuration are:
+
+- `FileSystemAccessApiWithFallback` - defaults to File System Access API, but falls back to download/upload pickers if not available
+- `FileSystemAccessApi` - uses File System Access API only. If not avaialable, pickers will throw `NotSupportedException`
+- `DownloadUpload` - uses download/upload pickers only.
+
+## Android
+
+Files picked from file pickers on Android are provided by the *Storage Access Framework API*. Due to its limitations, it is not possible to write to existing file in-place. Instead, Uno Platform creates a copy of the file in temporary storage and your changes are applied to this temporary file instead. When your file stream is then flushed, closed, or disposed of, the changes are written to the source file and the temporary file is discarded.
+
+## iOS
+
+iOS does not offer a built-in `FileSavePicker` alternative. Luckily it is possible to implement this functionality for example using a combination of a `FolderPicker` and `ContentDialog`.
+
+To provide your own custom implementation, create a class that implements the `IFileSavePickerExtension` which is only available on iOS. This class must have a `public` constructor with a `object` parameter. This will actually be an instance of `FileSavePicker` when invoked later. Then implement the `PickSaveFileAsync` method:
```

Comment:
You should also mention this .ctor constraint as xml documentation on the interface too for ease of access. Although it is more likely that they will discover it in this doc.

Reply:
- [ ] @MartinZikmund